### ISSUE TITLE
Help, when I finished the software installation, the first time I opened it, the first time I tried to generate images, I encountered this error.

### ISSUE THREAD

**Issue by wushuiliusuantong** [easydiffusion/easydiffusion#1499]
Error: Unexpected Read Error:
Error:undefined
EventInfo: {
    "reader": {}
}

Error: The model for Stable Diffusion has not been loaded yet! If you've tried to load it, please check the logs above this message for errors (while loading the model).

![Snipaste_2023-08-15_21-16-55](https://github.com/easydiffusion/easydiffusion/assets/110048278/4e1fda54-f075-4d5f-9834-1cf682658ab3)

Log:
Install dir: D:\EasyDiffusion\
D:\EasyDiffusion\installer_files\env\Library\bin\git.exe
git version 2.41.0.windows.1
D:\EasyDiffusion\installer_files\env\Library\bin\conda.bat
D:\EasyDiffusion\installer_files\env\Scripts\conda.exe
conda 22.11.1

"Easy Diffusion - v2"

"Easy Diffusion's git repository was already installed. Updating from main.."
HEAD is now at 6158f49 Workaround to fix the broken Rabbit Hole plugin
Already on 'main'
Your branch is up to date with 'origin/main'.
fatal: unable to access 'https://github.com/cmdr2/stable-diffusion-ui.git/': schannel: next InitializeSecurityContext failed: Unknown error (0x80092012)
493 File(s) copied
        1 file(s) copied.
        1 file(s) copied.
        1 file(s) copied.
        1 file(s) copied.
        1 file(s) copied.
        1 file(s) copied.
        1 file(s) copied.
        1 file(s) copied.
A subdirectory or file tmp already exists.

Hotfixed broken JSON file from OpenAI
"torch and torchvision have already been installed."
"sdkit is already installed."
sdkit version: 1.0.43
stable-diffusion version: 2.1.3
"rich has already been installed."
"Packages necessary for Easy Diffusion were already installed"
"Data files (weights) necessary for Stable Diffusion were already downloaded. Using the HuggingFace 4 GB Model."
"Data files (weights) necessary for GFPGAN (Face Correction) were already downloaded"
"Data files (weights) necessary for ESRGAN (Resolution Upscaling) x4plus were already downloaded"
"Data files (weights) necessary for ESRGAN (Resolution Upscaling) x4plus_anime were already downloaded"
"Data files (weights) necessary for the default VAE (sd-vae-ft-mse-original) were already downloaded"

"Easy Diffusion installation complete! Starting the server!"

PYTHONPATH=D:\EasyDiffusion\installer_files\env\lib\site-packages
D:\EasyDiffusion\installer_files\env\python.exe
C:\Users\Administrator\AppData\Local\Programs\Python\Python310\python.exe
C:\Users\Administrator\AppData\Local\Microsoft\WindowsApps\python.exe
Python 3.8.5
No module 'xformers'. Proceeding without it.
21:14:09.019 INFO MainThread started in D:\EasyDiffusion\stable-diffusion                                   server.py:19
21:14:09.024 INFO MainThread started at 08/15/23 21:14:09                                                   server.py:20
21:14:09.152 INFO MainThread Scanned 3 models. Nothing infected                                     model_manager.py:248
21:14:09.747 INFO MainThread Start new Rendering Thread on device: cuda:0                            task_manager.py:454
21:14:10.353 WARNING cuda:0 forcing full precision on this GPU, to avoid green images. GPU         device_manager.py:130
detected: NVIDIA GeForce GTX 1660 Ti
21:14:10.356 INFO cuda:0 Setting cuda:0 as active, with precision: full                            device_manager.py:134
21:14:10.358 INFO cuda:0 loading stable-diffusion model from                                              __init__.py:23
D:\EasyDiffusion\models\stable-diffusion\sd-v1-4.ckpt to device: cuda:0
21:14:14.218 INFO cuda:0 using config:                                                                    __init__.py:57
D:\EasyDiffusion\installer_files\env\lib\site-packages\sdkit\models\models_db\configs\v1-inference.yaml
21:14:14.235 INFO cuda:0 using attn_precision: fp32                                                       __init__.py:63
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
making attention of type 'vanilla' with 512 in_channels
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
making attention of type 'vanilla' with 512 in_channels
21:14:24.748 ERROR cuda:0 Error while loading stable-diffusion model:                                model_manager.py:45
D:\EasyDiffusion\models\stable-diffusion\sd-v1-4.ckpt
21:14:24.750 ERROR cuda:0 Error: Unable to load weights from pytorch checkpoint file for             model_manager.py:46
'D:\EasyDiffusion\profile/.cache\huggingface\hub\models--openai--clip-vit-large-patch14\snapshots\8d
052a0f05efbaefbc9e8786ba291cfdf93e5bff\pytorch_model.bin' at
'D:\EasyDiffusion\profile/.cache\huggingface\hub\models--openai--clip-vit-large-patch14\snapshots\8d
052a0f05efbaefbc9e8786ba291cfdf93e5bff\pytorch_model.bin'. If you tried to load a PyTorch model from
a TF 2.0 checkpoint, please set from_tf=True.
21:14:24.757 ERROR cuda:0 Consider removing the model from the model folder.                         model_manager.py:47
21:14:24.759 INFO cuda:0 loading vae model from                                                           __init__.py:23
D:\EasyDiffusion\models\vae\vae-ft-mse-840000-ema-pruned.ckpt to device: cuda:0
21:14:25.905 INFO cuda:0 loaded vae model from                                                            __init__.py:27
D:\EasyDiffusion\models\vae\vae-ft-mse-840000-ema-pruned.ckpt to device: cuda:0
21:16:43.717 INFO cuda:0 Session 1692105251756 starting task 2719394412432 on NVIDIA GeForce GTX     task_manager.py:312
1660 Ti
21:16:43.721 INFO cuda:0 request: {'guidance_scale': 7.5,                                                 renderer.py:51
 'height': 256,
 'hypernetwork_strength': 0,
 'negative_prompt': '',
 'num_inference_steps': 25,
 'num_outputs': 1,
 'preserve_init_image_color_profile': False,
 'prompt': 'a photograph of an astronaut riding a horse',
 'sampler_name': 'euler_a',
 'seed': 1629659,
 'width': 256}
21:16:43.726 INFO cuda:0 task data: {'block_nsfw': False,                                                 renderer.py:52
 'metadata_output_format': 'none',
 'output_format': 'jpeg',
 'output_quality': 75,
 'request_id': 2719394412432,
 'save_to_disk_path': None,
 'session_id': '1692105251756',
 'show_only_filtered_image': True,
 'stream_image_progress': False,
 'stream_image_progress_interval': 5,
 'upscale_amount': 4,
 'use_face_correction': None,
 'use_hypernetwork_model': None,
 'use_stable_diffusion_model': 'D:\\EasyDiffusion\\models\\stable-diffusion\\sd-v1-4.ckpt',
 'use_upscale': None,
 'use_vae_model': 'D:\\EasyDiffusion\\models\\vae\\vae-ft-mse-840000-ema-pruned.ckpt',
 'vram_usage_level': 'balanced'}
21:16:43.731 INFO cuda:0 Global seed set to 1629659                                                           seed.py:65
21:16:43.743 ERROR cuda:0 Traceback (most recent call last):                                         task_manager.py:346
  File "D:\EasyDiffusion\ui\easydiffusion\task_manager.py", line 336, in thread_render
    task.response = renderer.make_images(
  File "D:\EasyDiffusion\ui\easydiffusion\renderer.py", line 38, in make_images
    images, seeds = make_images_internal(req, task_data, data_queue, task_temp_images,
step_callback)
  File "D:\EasyDiffusion\ui\easydiffusion\renderer.py", line 59, in make_images_internal
    images, user_stopped = generate_images_internal(
  File "D:\EasyDiffusion\ui\easydiffusion\renderer.py", line 91, in generate_images_internal
    images = generate_images(context, callback=callback, **req.dict())
  File "D:\EasyDiffusion\installer_files\env\lib\site-packages\sdkit\generate\image_generator.py",
line 50, in generate_images
    raise RuntimeError(
RuntimeError: The model for Stable Diffusion has not been loaded yet! If you've tried to load it,
please check the logs above this message for errors (while loading the model).

21:16:43.834 INFO cuda:0 Session 1692105251756 task 2719394412432 failed!                            task_manager.py:355

On the Window10

**Comment by JeLuF** [comment]
The CLIP model file is broken. Please delete the directory 
`D:\EasyDiffusion\profile\.cache\huggingface\hub\models--openai--clip-vit-large-patch14\snapshots\8d
052a0f05efbaefbc9e8786ba291cfdf93e5bff\`
and restart Easy Diffusion. Easy Diffusion will attempt a new download of the model file.

In your logs, I can see that EasyDiffusion fails to communicate with github:
```
fatal: unable to access 'https://github.com/cmdr2/stable-diffusion-ui.git/': schannel: next InitializeSecurityContext failed: Unknown error (0x80092012)
```
This could indicate that there's a general issue with internet connectivity for EasyDiffusion. This can be caused by anti-malware, VPN tunnels, or ISP or government side network restrictions. During the restart of Easy Diffusion, please check whether the download from huggingface.co succeeds or whether it also encounters issues.

**Comment by ClockLoop** [comment]
> The CLIP model file is broken. Please delete the directory `D:\EasyDiffusion\profile\.cache\huggingface\hub\models--openai--clip-vit-large-patch14\snapshots\8d 052a0f05efbaefbc9e8786ba291cfdf93e5bff\` and restart Easy Diffusion. Easy Diffusion will attempt a new download of the model file.
> 
> In your logs, I can see that EasyDiffusion fails to communicate with github:
> 
> ```
> fatal: unable to access 'https://github.com/cmdr2/stable-diffusion-ui.git/': schannel: next InitializeSecurityContext failed: Unknown error (0x80092012)
> ```
> 
> This could indicate that there's a general issue with internet connectivity for EasyDiffusion. This can be caused by anti-malware, VPN tunnels, or ISP or government side network restrictions. During the restart of Easy Diffusion, please check whether the download from huggingface.co succeeds or whether it also encounters issues.

CANNOT COPY DATA OUT OF MEDIA TENSOR.
I have this problem almost weekly, and the only solution for me is to delete all the cache files and try to make easy diffusion download them all again, but easy diffustion will corrupt the files again at one point when it thinks they are missing again by re-downloading them and corrupting the good file.
It will do this after rebooting ED from a known good state.
I can only guess that the ED downloader is no so good, like doesn't it check the crc of the files before thinking they are bad?,  Does it have a retry method to acutally make sure the files it needs ARE indeed bad?
One time I checked out a file it says was bad, which I knew was good for a fact, I opened the file and it was a hugging face WEB PAGE in html format.  NOT the file it wanted (which was one of the gfpgan files which WAS good at one point before ED decided to re-download a webpage in page of the GFPGAN file thinking its download was good.
This then gives the "could not load data out of tendor" error.
This is reproducible, but only randomly.  Sometimes it won't do it, and sometimes it will corrupt its own downloads and then try to re-download the file.
We need a crc check system of some sort for these files since they are so big I understand its hard to make people wait to calculate a crc, but perhaps giving some of us who would rather have the program work, an option to VERIFY the files are in fact good and not a huggingface webpage for the file, instead of the actual file we want.

I use the beta so if you push out a downloader fix for this issue, I will be testing it.
Until then, I have tried to make the cache files and model files READ ONLY, but you force your program to error out if those files are read only, because you want to make the "place your diffusion files here.txt" file, and read only won't let it so you critital error instead of keep running.

So if we could get a downloader that has crc/verification of the file contents, and a way to NOT modify the cache and model files, that would likely eliminate these downloader issues.
So what sup what this ed downloader anyway?   Is is not so good?
I rarely have files download the html page instead of the actual file.  This could be a hugging face server issue, but a client side crc/ verification tool/method routine would kill this bug.
Gopod luck and if I can help you identify this downloader bug, send me  a message.
So far I can only fix it by making ED re-download everything repeatedly, deleting the model and cache files, again and again until FINALLY one of the download rounds makes all the files proper, and not a pseudo file with a hugging face html webpage inside it.

**Comment by ClockLoop** [comment]
The easy-diffusion/stable-diffusion/gfpgan/weights files are also getting corrupted, and I didn't realize this so I was not deleteing them along with the model, and cache files.
Finally...   that gave me a run around for a an hour.
So now, is there a list of ALL the files ED downloads after the program starts up?
 AFTER THIS POINT:
<details><summary>Details</summary>
<p>

Easy Diffusion installation complete, starting the server!



</p>
</details>

**Comment by ClockLoop** [comment]
Just now, ED tried to re-download a cache file that was already in the cache. But it tried anyway and was refused by my router configuration.

So ED threw an error and then just kept on going and rendered the image just fine.

At this point where it tries to re-download a file it already has which works, is when ED will download a new file in place of the already working file (perhaps it thinks its too old) what ever the reason it tried to download....
At which point it is possibly going to download a file that is a huggingface html file instead, I never asked it to download that already downloaded cache file and I was only able to stop it by fencing the pc from the 'net.
And ED seems to do this with all the other files it already has, perhaps for an update, but it eventually downloads a file incorrectly overwrites a good file (but older?), and boom.
'media tensor error' or some other bad data file error.


```shell
                                                        
19:33:42.873 INFO cuda:0 loading on diffusers                                                                                                          __init__.py:145
19:33:42.874 INFO cuda:0 using config:                                                                                                                 __init__.py:147
/home/user/Desktop/easy-diffusion/installer_files/env/lib/python3.8/site-packages/sdkit/models/models_db/configs/v1-inference.yaml                                  
19:33:42.883 INFO cuda:0 using attn_precision: fp16                                                                                                    __init__.py:156

'(ReadTimeoutError("HTTPSConnectionPool(host='huggingface.co', port=443): Read timed out. (read timeout=10)"), '(Request ID: 3b7e6c73-e8b4-417d-a02b-b8437d4b920c)')' thrown while requesting HEAD https://huggingface.co/openai/clip-vit-large-patch14/resolve/main/config.json
19:33:53.527 WARNING cuda:0 '(ReadTimeoutError("HTTPSConnectionPool(host='huggingface.co', port=443): Read timed out. (read  _http.py:271
timeout=10)"), '(Request ID: 3b7e6c73-e8b4-417d-a02b-b8437d4b920c)')' thrown while requesting HEAD                                       
https://huggingface.co/openai/clip-vit-large-patch14/resolve/main/config.json                                                            

You have disabled the safety checker for <class 'diffusers.pipelines.stable_diffusion.pipeline_stable_diffusion_img2img.StableDiffusionImg2ImgPipeline'> by passing `safety_checker=None`. Ensure that you abide to the conditions of the Stable Diffusion license and do not expose unfiltered results in services or applications open to the public. Both the diffusers team and Hugging Face strongly recommend to keep the safety filter enabled in all public facing circumstances, disabling it only for use-cases that involve analyzing network behavior or auditing its results. For more information, please have a look at https://github.com/huggingface/diffusers/pull/254 .

You have disabled the safety checker for <class 'diffusers.pipelines.stable_diffusion.pipeline_stable_diffusion_inpaint_legacy.StableDiffusionInpaintPipelineLegacy'> by passing `safety_checker=None`. Ensure that you abide to the conditions of the Stable Diffusion license and do not expose unfiltered results in services or applications open to the public. Both the diffusers team and Hugging Face strongly recommend to keep the safety filter enabled in all public facing circumstances, disabling it only for use-cases that involve analyzing network behavior or auditing its results. For more information, please have a look at https://github.com/huggingface/diffusers/pull/254 .

19:34:27.141 INFO cuda:0 Global seed set to 42                                                                              seed.py:65
19:34:27.144 INFO cuda:0 Using sampler: EulerAncestralDiscreteScheduler {                                       image_generator.py:330
  "_class_name": "EulerAncestralDiscreteScheduler",                                                                                   
  "_diffusers_version": "0.19.2",                                                                                                     
  "beta_end": 0.012,                                                                                                                  
  "beta_schedule": "scaled_linear",                                                                                                   
  "beta_start": 0.00085,                                                                                                              
  "clip_sample": false,                                                                                                               
  "num_train_timesteps": 1000,                                                                                                        
  "prediction_type": "epsilon",                                                                                                       
  "set_alpha_to_one": false,                                                                                                          
  "steps_offset": 1,                                                                                                                  
  "timestep_spacing": "linspace",                                                                                                     
  "trained_betas": null                                                                                                               
}                                                                                                                                     
 because of euler_a                                                                                                                   
19:34:27.146 INFO cuda:0 Applying Embeddings...                                                                 image_generator.py:509
19:34:27.147 INFO cuda:0 Applying tiling settings                                                               image_generator.py:360
19:34:27.150 INFO cuda:0 Parsing the prompt...                                                                  image_generator.py:404
19:34:27.150 INFO cuda:0 compel is ready                                                                        image_generator.py:408
19:34:27.211 INFO cuda:0 Made prompt embeds                                                                     image_generator.py:430
19:34:27.252 INFO cuda:0 Made negative prompt embeds                                                            image_generator.py:433
19:34:27.276 INFO cuda:0 Done parsing the prompt                                                                image_generator.py:439
```

**Comment by ClockLoop** [comment]
I have also found that the UI folder needs to be deleted when running into corrupt files, the ui seems to remember the location of corrupt files? when the interface selects them, even after rebooting the server, ..??  odd.?  perhaps something else is going on, but nukin the ui folder before starting the server helped me once..   
So my list grows.
The 'huggingface' folder located under your operating systems TEMP or cache location.
The 'model' folder located inside 'easy-diffusions' folder.
And the 'stable-diffusion' located inside 'easy-diffusions' folder.  (should this be done, will it redownload them?)
The 'ui' folder located inside 'easy-diffusion' folder. (does the ui cache any files, like gfpgan or realesrgan?)
All need to be deleted in the case of corrupt files, not just the cache.

**Comment by JeLuF** [comment]
> ```shell
> '(ReadTimeoutError("HTTPSConnectionPool(host='huggingface.co', port=443): Read timed out. (read timeout=10)"), '(Request ID: 3b7e6c73-e8b4-417d-a02b-b8437d4b920c)')' thrown while requesting HEAD https://huggingface.co/openai/clip-vit-large-patch14/resolve/main/config.json
> 19:33:53.527 WARNING cuda:0 '(ReadTimeoutError("HTTPSConnectionPool(host='huggingface.co', port=443): Read timed out. (read  _http.py:271
> timeout=10)"), '(Request ID: 3b7e6c73-e8b4-417d-a02b-b8437d4b920c)')' thrown while requesting HEAD                                       
> https://huggingface.co/openai/clip-vit-large-patch14/resolve/main/config.json 
> ```

A HEAD request is not a download. It just checks how old the source file is in order to compare it to the local file. Only if the source file would be newer than the local file, it would update the cache.

> And the 'stable-diffusion' located inside 'easy-diffusions' folder. (should this be done, will it redownload them?)

Deleting this folder will break your EasyDiffusion installation. It will not be redownloaded.

> The 'ui' folder located inside 'easy-diffusion' folder. (does the ui cache any files, like gfpgan or realesrgan?)

There is no cached data in the UI folder. It only contains program code. If nuking this folder has helped, it was probably the restart per se that fixed it, not the deletion of that folder.
